crates/gossip/src/initiate.rs

```diff
@@ -356,9 +354,8 @@ mod tests {
         let harness = Harness::create().await;
 
```

Comment:
This is unrelated to the PR but I'm wondering whether we should also assert in this test that `harness.local_agent_store.get_all().await.len() == 0`?